lib/instrumentation/index.js

```diff
@@ -106,27 +90,44 @@ Instrumentation.prototype._patchModule = function (exports, name, version, enabl
 }
 
 Instrumentation.prototype.addEndedTransaction = function (transaction) {
+  var agent = this._agent
+
   if (this._started) {
-    var queue = this._queue
+    var payload = agent._filters.process(transaction._encode()) // TODO: Update filter to expect this format
+    if (!payload) return agent.logger.debug('transaction ignored by filter %o', {id: transaction.id})
+    truncate.transaction(payload)
+    agent.logger.debug('sending transaction %o', {id: transaction.id})
+    if (agent._apmServer) agent._apmServer.sendTransaction(payload)
```

Comment:
Is there any way for `agent._apmServer` to be null when `this._started` is true? 🤔

Reply:
No you're right - if `this._started === true`, then we'll also have an `agent._apmServer`. Would that be a better check? The current check is guaranteed to always work, whereas the conditions of the other might change in the future right?

Reply:
I just made that comment, because this check is already in an outer `if` block checking `this._started`, so it seemed redundant. Github cuts off the preview just after that line though. 😅

Reply:
Oh yes... I'll remove it

Reply:
Fixed